sdk/data/azcosmos/partition_key_definition.go

```diff
@@ -3,11 +3,84 @@
 
 package azcosmos
 
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type PartitionKeyKind string
+
+const (
+	Hash      PartitionKeyKind = "Hash"
+	MultiHash PartitionKeyKind = "MultiHash"
```

Comment:
For other "enums", we usually prefix the name of the variant with the type name, so these should be:

```suggestion
	PartitionKeyKindHash      PartitionKeyKind = "Hash"
	PartitionKeyKindMultiHash PartitionKeyKind = "MultiHash"
```

It's more verbose, for sure, but it's consistent with what we've done in other places.